src/gpuarray_array.c

```diff
@@ -1008,15 +1009,18 @@ int GpuArray_concatenate(GpuArray *r, const GpuArray **as, size_t n,
 
   res_off = r->offset;
   res_dims = r->dimensions;
+  res_flags = r->flags;
+  r->flags &= ~(GA_C_CONTIGUOUS|GA_F_CONTIGUOUS);
```

Comment:
This is a corner case that doesn't happen anywhere in practice because all operations set the flags correctly on the arrays.  Here we muck with the array properties directly and therefore we have to adjust the flags to match.  In the interest of code size and speed I err on the safe side and just always say that the array is not contiguous because the non-contiguous code is safe in the contiguous case and the contiguous case is rare in this instance.

I oppose adding code to handle what is technically a violation of the interface.